app/controllers/auth_key_pair_cloud_controller.rb

```diff
@@ -112,21 +112,17 @@ def create
       ext_management_system = find_by_id_filtered(ManageIQ::Providers::CloudManager, options[:ems_id])
       kls = kls.class_by_ems(ext_management_system)
       if kls.is_available?(:create_key_pair, ext_management_system, options)
-        begin
-          kls.create_key_pair(ext_management_system, options)
-          add_flash(_("Creating %{model} %{name}") % {
-            :model => ui_lookup(:table => 'auth_key_pair_cloud'),
-            :name  => options[:name]})
-        rescue => ex
-          add_flash(_("Unable to create %{model} %{name}. %{error}") % {
-            :model => ui_lookup(:table => 'auth_key_pair_cloud'),
-            :name  => options[:name],
-            :error => get_error_message_from_fog(ex.to_s)}, :error)
+        task_id = kls.create_key_pair_queue(session[:userid], ext_management_system, options)
+
+        add_flash(_("%{model} creation failed: Task start failed: ID [%{id}]") %
+                  {:model => ui_lookup(:table => 'auth_key_pair_cloud'),
+                   :id    => task_id.to_s}, :error) unless task_id.kind_of?(Integer)
```

Comment:
I know this code has been there before, but instead of `ui_lookup(:table => 'auth_key_pair_cloud')`, please use
just plain `Key Pair` string:

```ruby
_("Key Pair creation failed: Task start failed: ID [%{id}]") % {:id    => task_id.to_s}}
```